core/src/main/java/io/jenkins/pluginhealth/scoring/probes/ThirdPartyRepositoryDetectionProbe.java

```diff
@@ -0,0 +1,111 @@
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(value = ThirdPartyRepositoryDetectionProbe.ORDER)
+public class ThirdPartyRepositoryDetectionProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ThirdPartyRepositoryDetectionProbe.class);
+    public static final int ORDER = SCMLinkValidationProbe.ORDER + 100;
+    public static final String KEY = "third-party-repository-detection-probe";
+    final String hostName = "https://repo.jenkins-ci.org";
+//    final String parentPom = "https://raw.githubusercontent.com/jenkinsci/plugin-pom/master/pom.xml";
+    final String parentPom = "https://github.com/jenkinsci/plugin-pom/blob/master/pom.xml";
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        MavenXpp3Reader mavenReader = new MavenXpp3Reader();
+        Set<Repository> allRepositories = new HashSet<>();
+
+        try {
+            InputStream inputStream = new FileInputStream(context.getScmRepository() + "/pom.xml");
+            Reader reader = new InputStreamReader(inputStream, "UTF-8");
+            Model model = mavenReader.read(reader);
+            allRepositories.addAll(model.getRepositories());
+            allRepositories.addAll(model.getPluginRepositories());
+
+            if (!model.getParent().getRelativePath().isBlank()) {
+                Model parentPomModel = parsePomFromUrl(model.getParent().getRelativePath());
+                allRepositories.addAll(parentPomModel.getRepositories());
+                allRepositories.addAll(parentPomModel.getPluginRepositories());
+            }
+            for (Repository repository : allRepositories) {
+                if (!repository.getUrl().startsWith(hostName)) {
+                    return ProbeResult.failure(KEY, "Third party repositories detected in the plugin");
+                }
+            }
+        } catch (FileNotFoundException e) {
+            LOGGER.error("File not found at {}", plugin.getName());
+            return ProbeResult.error(KEY, e.getMessage());
+        } catch (XmlPullParserException e) {
+            LOGGER.error("Pom file could not be parsed at {}", plugin.getName());
+            return ProbeResult.error(KEY, e.getMessage());
+        } catch (IOException e) {
+            LOGGER.error("File reading exception at {}", plugin.getName());
+            return ProbeResult.error(KEY, e.getMessage());
+
+        }
+        return allRepositories.size() > 0 ? ProbeResult.success(KEY, "The plugin has no third party repositories")
+            : ProbeResult.failure(KEY, "No repositories detected");
+    }
+
+    @Override
+    public String key() {
+        return KEY;
+    }
+
+    @Override
+    public String getDescription() {
+        return "Detects third-party repositories in a plugin.";
+    }
+
+    @Override
+    public String[] getProbeResultRequirement() {
+        return new String[] { SCMLinkValidationProbe.KEY};
```

Comment:
Have to choose, space or no space before/after `{/}` but not both.

```suggestion
        return new String[] { SCMLinkValidationProbe.KEY };
```